dot/state/block.go

```diff
@@ -438,6 +442,7 @@ func (bs *BlockState) SetFinalizedHash(hash common.Hash, round, setID uint64) er
 			return err
 		}
 	}
+	bs.pruneKeyCh <- pruned
```

Comment:
I think this should be moved inside the for loop, since we only want to delete storage tries that are no longer part of the finalized chain (similarly to the discussion we had in standup for deleting the blocks)